docs/deployment.md

```diff
@@ -0,0 +1,31 @@
+
+
+
+
+![image](images/ci_cd_github_big_picture_public.drawio.png)
+
+
+# Helm Chart
+
+![image](images/external-helm-chart.drawio.png)
+
+There is a sample chart. https://github.com/it-at-m/helm-charts/tree/main/charts/sps-sample . It consists of subcharts. Each subchart is a different module for example frontend, backend, eai.  Each module was created with `helm create` and some adaptions. The Chart.yml includes the external dependency of the apigateway. The main config is done in the values.yml of the parent chart. In the section refarch-gateway, frontend, backend you can set the config of the subchart. 
+
+
+
+
+# Deployment
+
+At the end of a sprint a developer builds a release in the source code project. It needs to be done for each folder frontend/backend/eai separately. Then the devops engineer needs to update the chart-version and app-version of the helm chart in the Chart.yaml.
+
+Internal the quay syncs the docker images with the external github image registry. The github api has a pull limit so we need to cache the images internally.
+
+![image](images/internal-chart.drawio.png)
+
+The internal gitlab repo you can config your application, for example the trust store, ingress (openshift route), single sign on, database config, environment configs. The most config is comment out, so you need to comment it in. Here you see an example https://git.muenchen.de/ccse/cicd/sps-github
+
+
+You have two possibilities for the image rollout. 
+-	For the dev environment, you want to have an auto rollout. We create a image stream. The image stream has a link to the quay. In the deployment there is a special annotation set. Openshift then triggers automatically trigger a new rollout if a new image is available.  https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/images/triggering-updates-on-imagestream-changes#triggering-updates-on-imagestream-changes
```

Comment:
Consider to change Quay with "internal image repository", as well replacing "Openshift" with "internal kubernetes platform".

Reply:
```suggestion
-	For the dev environment, you want to have an auto rollout. We create a image stream. The image stream has a link to the internal image registry (Quay). In the deployment there is a special annotation set. The internal kubernetes platform (Openshift) then triggers automatically trigger a new rollout if a new image is available.  https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/images/triggering-updates-on-imagestream-changes#triggering-updates-on-imagestream-changes
```

Reply:
wenn wir jetzt in Github zur Quay "internal image repository" und Openshift (internal kubernetes plattform) nennen, kennt sich keiner mehr aus. Ich finde wir sollten für externe Doku dieselben Begriffe nehmen

Reply:
Gerne Quay und Openshift in Klammern. Aber wenn du nur Quay und Openshift verwendest, ist es nicht mehr allgemeingültig für alle (auch Nicht-LHMler) und wenn wir intern was ändern, musst du u.U. nur deswegen die Doku ändern.